### PR TITLE
package upgrades

### PR DESCRIPTION
**Pull request by vaishnavi-2301**
Update compileSdkVersion and dependencies.

### CODE REVIEW

**Comment by vaishnavi-2301** [comment]
@diegotori please let me know if there's a requirement of updating the files in android folder according to the new migration guide https://docs.flutter.dev/release/breaking-changes/flutter-gradle-plugin-apply

will be happy to contribute!!

**Comment by diegotori** [comment]
@vaishnavi-2301 please address the lint issues.

Also, since this library is merely a UI layer over `video_player`, the only thing that needs migrating is the Android example app.

**Comment by vaishnavi-2301** [comment]
@diegotori  Hey, addressed the lint issues. kindly review and let me know if there's anything required.

**Review by diegotori — CHANGES_REQUESTED** [review]
Please ensure that the Android example app is closely configured based on [this example](https://github.com/fluttercommunity/wakelock_plus/tree/main/wakelock_plus/example/android) from `wakelock_plus`.

The reason being is that I recently had to shake some bugs when actually building the migrated Android app. The example that I linked you to should be the gold standard in terms of that.

Mainly, it involves building the app against Java 17, but please cross-check your changes against that example.

Thanks in advance.

File: example/android/app/build.gradle [hunk 1]
```diff
@@ -21,24 +23,33 @@ if (flutterVersionName == null) {
...
         main.java.srcDirs += 'src/main/kotlin'
     }
 
     defaultConfig {
-        // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
-        applicationId "com.example.example"
-        minSdkVersion 21
-        targetSdkVersion 33
+        applicationId  "com.example.example"
+        minSdk 24
```
Comment (diegotori) [inline example/android/app/build.gradle:51]: @vaishnavi-2301 Is there any reason why `minSdk` on Android needs to be set at `24`?

Unless you cite where it needs to be set that high, please set it to `21`, which is the bare minimum that Flutter now supports. Thanks.

**Review by diegotori — CHANGES_REQUESTED** [review]

File: example/android/app/build.gradle (hunk 1 above)
Comment (vaishnavi-2301) [reply example/android/app/build.gradle:51]: Okay will change it to the 21, i updated it according to your reference of wakelock_plus.

**Comment by diegotori** [comment]
@vaishnavi-2301 added in version `1.8.2`.

**Pull request merged**